src/tabletoolbar.js

```diff
@@ -0,0 +1,194 @@
+		 */
+		this._balloon = this.editor.plugins.get( 'ContextualBalloon' );
+
+		/**
+		 * A `ToolbarView` instance used to display the buttons specific for table editing.
+		 *
+		 * @protected
+		 * @type {module:ui/toolbar/toolbarview~ToolbarView}
+		 */
+		this._toolbar = new ToolbarView();
+
+		// Add buttons to the toolbar.
+		this._toolbar.fillFromConfig( toolbarConfig, editor.ui.componentFactory );
+
+		// Show balloon panel each time table widget is selected.
+		this.listenTo( editor.editing.view, 'render', () => {
+			this._checkIsVisible();
+		} );
+
+		// There is no render method after focus is back in editor, we need to check if balloon panel should be visible.
+		this.listenTo( editor.ui.focusTracker, 'change:isFocused', () => {
+			this._checkIsVisible();
+		}, { priority: 'low' } );
+	}
+
+	/**
+	 * Checks whether the toolbar should show up or hide depending on the current selection.
+	 *
+	 * @private
+	 */
+	_checkIsVisible() {
+		const editor = this.editor;
+
+		if ( !editor.ui.focusTracker.isFocused ) {
+			this._hideToolbar();
+		} else {
+			if ( isTableWidgetSelected( editor.editing.view.document.selection ) ) {
+				this._showToolbar();
+			} else {
+				this._hideToolbar();
+			}
+		}
+	}
+
+	/**
+	 * Shows the {@link #_toolbar} in the {@link #_balloon}.
+	 *
+	 * @private
+	 */
+	_showToolbar() {
+		const editor = this.editor;
+
+		if ( this._isVisible ) {
+			repositionContextualBalloon( editor );
+		} else {
+			if ( !this._balloon.hasView( this._toolbar ) ) {
+				this._balloon.add( {
+					view: this._toolbar,
+					position: getBalloonPositionData( editor ),
+					balloonClassName
+				} );
+			}
+		}
+	}
+
+	/**
+	 * Removes the {@link #_toolbar} from the {@link #_balloon}.
+	 *
+	 * @private
+	 */
+	_hideToolbar() {
+		if ( !this._isVisible ) {
+			return;
+		}
+
+		this._balloon.remove( this._toolbar );
+	}
+
+	/**
+	 * Returns `true` when the {@link #_toolbar} is the visible view in the {@link #_balloon}.
+	 *
+	 * @private
+	 * @type {Boolean}
+	 */
+	get _isVisible() {
+		return this._balloon.visibleView == this._toolbar;
+	}
+}
+
+/**
+ * Items to be placed in the table toolbar.
+ * This option is used by the {@link module:table/tabletoolbar~TableToolbar} feature.
+ *
+ * Assuming that you use the {@link module:table/tableui~TableUI} feature below toolbar items will be available
+ * in {@link module:ui/componentfactory~ComponentFactory}:
+ *
+ * * `'tableRow'`,
+ * * `'tableColumn'`,
+ * * `'mergeCell'`,
+ * * `'splitCell'`,
+ *
+ * so you can configure the toolbar like this:
+ *
+ *		const tableConfig = {
+ *			toolbar: [ 'tableRow', 'tableColumn', 'mergeCell', 'splitCell' ]
+ *		};
+ *
+ * Of course, the same buttons can also be used in the
+ * {@link module:core/editor/editorconfig~EditorConfig#toolbar main editor toolbar}.
+ *
+ * Read more about configuring toolbar in {@link module:core/editor/editorconfig~EditorConfig#toolbar}.
+ *
+ * @member {Array.<String>} module:table~TableConfig#toolbar
```

Comment:
Where's the @interface TableConfig defined BTW because I couldn't find it? I can't find these docs after building under `api/module_core_editor_editorconfig-EditorConfig.html`